pkg/inclusion/nmt_caching_test.go

```diff
@@ -0,0 +1,178 @@
+package inclusion
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/testutil/coretestutil"
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/pkg/consts"
+	"github.com/tendermint/tendermint/pkg/da"
+	"github.com/tendermint/tendermint/pkg/wrapper"
+)
+
+func TestWalkCachedSubTreeRoot(t *testing.T) {
+	// create the first main tree
+	strc := newSubTreeRootCacher()
+	oss := uint64(8)
+	tr := wrapper.NewErasuredNamespacedMerkleTree(oss, nmt.NodeVisitor(strc.Visit))
+	d := []byte{0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 3, 4, 5, 6, 7, 8}
+	for i := 0; i < 8; i++ {
+		tr.Push(d, rsmt2d.SquareIndex{
+			Axis: uint(rsmt2d.Row),
+			Cell: uint(i),
+		})
+	}
+	highestRoot := tr.Root()
+
+	// create a small sub tree
+	subTree1 := wrapper.NewErasuredNamespacedMerkleTree(oss)
+	for i := 0; i < 2; i++ {
+		subTree1.Push(d, rsmt2d.SquareIndex{
+			Axis: uint(rsmt2d.Row),
+			Cell: uint(i),
+		})
+	}
+	shortSTR := subTree1.Root()
+
+	// create a larger sub tree root
+	subTree2 := wrapper.NewErasuredNamespacedMerkleTree(oss)
+	for i := 0; i < 4; i++ {
+		subTree2.Push(d, rsmt2d.SquareIndex{
+			Axis: uint(rsmt2d.Row),
+			Cell: uint(i),
+		})
+	}
+	tallSTR := subTree2.Root()
```

Comment:
[nit][optional] consistent naming

```suggestion
	// create a tall sub tree root
	tallSubTree := wrapper.NewErasuredNamespacedMerkleTree(oss)
	for i := 0; i < 4; i++ {
		tallSubTree.Push(d, rsmt2d.SquareIndex{
			Axis: uint(rsmt2d.Row),
			Cell: uint(i),
		})
	}
	tallSTR := tallSubTree.Root()
```